homeassistant/components/automation/__init__.py

```diff
@@ -14,6 +14,7 @@
     CONF_PLATFORM,
     CONF_ZONE,
     EVENT_AUTOMATION_TRIGGERED,
+    EVENT_AUTOMATION_UPDATED,
```

Comment:
description talks about reloaded, here I see updated. I think that reloaded is a better description, because when we talk input_X, zones etc, storage based ones are updated right away, and we would not fire the event then ?

Reply:
I had EVENT_AUTOMATION_RELOADED, however, looking at the code base, that term is never used. hence I changed it to UPDATED, also because:

My PR initially started out as a reloaded for automations (with the goal for others as well) quickly realizing that won't work for collections, as they don't "reload", but still change.

See 7207a3f for zone, which uses a storage-based collection.

Generally, thinking we might need UPDATED & RELOADED both? As a reloaded does not change the collection if nothing is updated?

The goal in general, it to be able to hook up into the fact something changed you want to trigger on. E.g., logging, messaging, automations.

Reply:
Updated the OP to match the current state of the PR, however, that may change... (I made it a draft to get input along the way).

Reply:
So a change to a zone in a collection will only update that one zone. It won't even unload it, just apply as we go. Reloading YAML will reload all YAML based entities. So creating a generic event is tricky, because reloading automations doesn't mean that an automation has been updated.

Reply:
So considering that, should it be split into both an UPDATED (from the collection perspective) and a RELOADED?

Reply:
We discussed this a bit offline. Most things are actually captured as entities and so can use state changes. 

Except for automations and scenes.

Reply:
Yep! Let's adjust and get things moving 👍 

Thanks for your insights!